INT-481 Connection String to add PGOPTIONS (#242)
metalbear-co/docs: docs/sharing-the-cluster/db-branching-advanced-config.md

`dump_args` is only supported for MySQL and PostgreSQL. MSSQL, MongoDB, and Redis branches use their own internal dump mechanisms and do not support this field.
{% endhint %}

## Connection Settings (PostgreSQL)

> `connection_settings` is a map of PostgreSQL settings that mirrord applies to every connection it opens to the source database while building the branch. Each entry is sent at connection startup, so it is in effect before any schema dump or data copy runs.

Reviewer: ```suggestion
`connection_settings` is a map of PostgreSQL settings that mirrord applies to every connection it opens to the source database while building the branch. Each entry is set before any schema dump or data copy runs.
```

> For example, this can be used for a [Row-Level Security (RLS)](https://www.postgresql.org/docs/current/ddl-rowsecurity.html) policy that reads `current_setting('...')`. Without that setting, the copy fails with:

Reviewer: ```suggestion
Any PostgreSQL session variable works here (e.g. `role`, `search_path`, custom app settings). For example, if your source database uses [Row-Level Security (RLS)](https://www.postgresql.org/docs/current/ddl-rowsecurity.html) with a policy that reads `current_setting('...')`, mirrord's connection will fail without that setting.
```

```
ERROR: unrecognized configuration parameter "app.product_id"
```

Setting it through `connection_settings` makes the copy read past the policy:

```json
{
  "feature": {
    "db_branches": [
      {
        "type": "pg",
        "version": "16",
        "connection": { "url": "DATABASE_URL" },
        "connection_settings": {
          "app.product_id": "123456"
        },
        "copy": {
          "mode": "schema",
          "tables": {
            "kudos": { "filter": "product_id = 123456" }
          }
        }
      }
    ]
  }
}
```

> The settings affect only the connection mirrord uses to read the source - they are never written into the branch. Any PostgreSQL session GUC works here, not just RLS variables (for example `role` or `search_path`).

Reviewer: ```suggestion
These settings only apply while mirrord reads from the source, they are never written into the branch itself.
```

Reviewer: ill mov the last sentence to the beginning
